downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -53,11 +53,12 @@ private fun findGitOrSubmoduleDir(workingDirOrFile: File): Repository {
 open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(workingDir, vcsType) {
     companion object : Logging
 
-    private val repo = findGitOrSubmoduleDir(workingDir)
+    private val repo: Repository
+        get() = findGitOrSubmoduleDir(workingDir)
 
-    override fun isValid(): Boolean = repo.objectDatabase?.exists() == true
+    override fun isValid(): Boolean = repo.use { it.objectDatabase?.exists() == true }
 
-    override fun isShallow(): Boolean = repo.directory?.resolve("shallow")?.isFile == true
+    override fun isShallow(): Boolean = repo.use { it.directory?.resolve("shallow")?.isFile == true }
```

Comment:
Consider factoring out `repo.use` to a function.

Reply:
Sorry, no. I'm not going to put an 8-letters-call to a function (whose name would probably longer).